cunumeric/module.py

```diff
@@ -3202,8 +3326,13 @@ def _contract(
 
 
 def einsum(
-    expr, *operands, out=None, dtype=None, casting="safe", optimize=False
-):
+    expr: str,
+    *operands: ndarray,
```

Comment:
here's another place where cuNumeric's definition of `array_like` would be necessary. I don't think `operands` are `ndarray`s yet, but they are all `cunumeric.array_like`s. (note that the conversion is done at line 3396.)